crates/spidermonkey-embedding-splicer/src/lib.rs

```diff
@@ -144,6 +148,11 @@ impl Guest for SpidermonkeyEmbeddingSplicerComponent {
             },
         )) = decode(&engine)
         {
+            // merge the imports from the engine with the imports from the guest content
+            for (k, _) in &engine_resolve.worlds[engine_world].imports {
+                guest_imports.push(engine_resolve.name_world_key(k));
+            }
```

Comment:
I was taking a stab at this afternoon and found that this returned all the imports provided in the `wit` definition and not just the ones actively used by the engine who's list I was able to get with the following code snippet.

```rust
// Componentize the Starlingmonkey engine so that we can pull in the imports that are actually used.
let bytes = include_bytes!("../../../StarlingMonkey/host-apis/wasi-0.2.0/preview1-adapter-release/wasi_snapshot_preview1.wasm");
let component = wit_component::ComponentEncoder::default()
    .adapter("wasi_snapshot_preview1", bytes)
    .map_err(|e| e.to_string())?
    .module(&engine)
    .map_err(|e| e.to_string())?
    .encode()
    .map_err(|e| e.to_string())?;
let decoded = wit_component::decode(&component).unwrap();
let resolve = decoded.resolve();
let packages = decoded.package();
let world_id = resolve.select_world(packages, None).unwrap();
let world = &resolve.worlds[world_id];
// merge the imports actually used by the engine with the imports from the guest content.
for (import_key, _) in world.imports.iter() {
    guest_imports.push(resolve.name_world_key(import_key));
}
```